### PR TITLE
Added intro and validation tables to extended components

### PR DESCRIPTION
**Pull request by Meggin**
This pull request adds a new intro table and validation error table to each extended component specification.

The validation error tables have links to the validation errors doc which is currently awaiting merge in the /docs repo. These links will be broken until the doc is live.

### CODE REVIEW

File: extensions/amp-access/amp-access-spec.md [hunk 1]
```diff
@@ -18,9 +18,43 @@ limitations under the License.
...
-AMP Access or “AMP paywall and subscription support” provides control to publishers over what content can be accessed by a reader and with what restrictions,  based on the reader’s subscription status, number of views and so on.
+<!---
+More validations need to be added once implemented;
+this table will change soon.
+-->
+
+<table>
+  <tr>
+    <th width="40%"><strong>Validation Error</strong></th>
+    <th>Description</th>
```
Comment (Gregable) [inline extensions/amp-access/amp-access-spec.md]: Another important one is MANDATORY_ATTR_MISSING or INVALID_ATTR_VALUE for the attribute `type="application/json"`

Comment (Gregable) [inline extensions/amp-access/amp-access-spec.md]: Actually, this is confusing. There are two tags, the required .js (for which this doesn't apply) and the json config script tag (for which this does).

Comment (Gregable) [inline extensions/amp-access/amp-access-spec.md]: The other possible one is INVALID_ATTR_VALUE for the script `src` attribute.

File: extensions/amp-dynamic-css-classes/amp-dynamic-css-classes.md [hunk 2]
```diff
@@ -16,8 +16,38 @@ limitations under the License.
...
+    <td width="40%"><strong>Required Script</strong></td>
+    <td><code>&lt;script async custom-element="amp-dynamic-css-classes" src="https://cdn.ampproject.org/v0/amp-dynamic-css-classes-0.1.js">&lt;/script></code></td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Examples</strong></td>
+    <td>None</td>
+  </tr>
+</table>
+
+The following lists validation errors specific to the `amp-dynamic-css-classes` tag
```
Comment (Gregable) [inline extensions/amp-dynamic-css-classes/amp-dynamic-css-classes.md]: Unlike the other custom-elements, this script doesn't actually introduce a new tag. It's perhaps poorly named as a custom-element. Specifically, there is no "`amp-dynamic-css-classes` tag". Thus REQUIRED_SCRIPT is also not a possible error.

File: extensions/amp-facebook/amp-facebook.md [hunk 3]
```diff
@@ -16,9 +16,44 @@ limitations under the License.
...
+    <td><code>&lt;script async custom-element="amp-facebook" src="https://cdn.ampproject.org/v0/amp-facebook-0.1.js">&lt;/script></code></td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Examples</strong></td>
+    <td><a href="https://github.com/ampproject/amphtml/blob/master/examples/facebook.amp.html">facebook.amp.html</a></td>
+  </tr>
+</table>
 
-Displays a Facebook Post or Video. 
+**Note:** This tag is currently stable; however, it doesn't validate.
```
Comment (Gregable) [inline extensions/amp-facebook/amp-facebook.md]: This is outdated. The tag validates. Same above on line 26

File: extensions/amp-font/amp-font.md [hunk 4]
```diff
@@ -16,8 +16,50 @@ limitations under the License.
...
+    <td><a href="https://github.com/ampproject/amphtml/blob/master/examples/font.amp.html">font.amp.html</a></td>
+  </tr>
+</table>
+
+The following lists validation errors specific to the `amp-font` tag
+(see also `amp-font` in the [AMP validator specification](https://github.com/ampproject/amphtml/blob/master/validator/validator.protoascii):
+
+<table>
+  <tr>
+    <th width="40%"><strong>Validation Error</strong></th>
```
Comment (Gregable) [inline extensions/amp-font/amp-font.md]: optional. Unsure if worth mentioning, but the timeout attribute value must be a positive integer or you'll get INVALID_ATTR_VALUE

File: extensions/amp-iframe/amp-iframe.md [hunk 5]
```diff
@@ -16,7 +16,56 @@ limitations under the License.
...
+    <td><a href="https://github.com/ampproject/amphtml/blob/master/examples/everything.amp.html">everything.amp.html</a></td>
+  </tr>
+</table>
+
+The following lists validation errors specific to the `amp-iframe` tag
+(see also `amp-iframe` in the [AMP validator specification](https://github.com/ampproject/amphtml/blob/master/validator/validator.protoascii):
+
+<table>
+  <tr>
+    <th width="40%"><strong>Validation Error</strong></th>
```
Comment (Gregable) [inline extensions/amp-iframe/amp-iframe.md]: optional, similar to above, there are several attribute values that have specific rules or will trigger an INVALID_ATTR_VALUE error

File: extensions/amp-mustache/amp-mustache.md [hunk 6]
```diff
@@ -16,7 +16,50 @@ limitations under the License.
...
+    <td width="40%"><strong>Required Script</strong></td>
+    <td><code>&lt;script async custom-element="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.1.js">&lt;/script></code></td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Examples</strong></td>
+    <td>None</td>
+  </tr>
+</table>
+
+The following lists validation errors specific to the `amp-mustache` tag
```
Comment (Gregable) [inline extensions/amp-mustache/amp-mustache.md]: This one also doesn't have an exact matching tag. Instead, the tag is `<template type="amp-mustache">`.

File: extensions/amp-mustache/amp-mustache.md [hunk 7]
```diff
@@ -16,7 +16,50 @@ limitations under the License.
...
+(see also `amp-mustache` in the [AMP validator specification](https://github.com/ampproject/amphtml/blob/master/validator/validator.protoascii):
```
Comment (Gregable) [inline extensions/amp-mustache/amp-mustache.md]: see `template`

File: extensions/amp-mustache/amp-mustache.md [hunk 8]
```diff
@@ -16,7 +16,50 @@ limitations under the License.
...
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#attribute-contains-template-syntax">TEMPLATE_IN_ATTR_NAME</a></td>
+    <td>Mustache templates must be within the <code>amp-mustache</code> tag. Error thrown when templating syntax found in an attribute in another tag.</td>
+  </tr>
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#attribute-contains-unescaped-template-syntax">UNESCAPED_TEMPLATE_IN_ATTR_VALUE</a></td>
+    <td>Mustache templates must be within the <code>amp-mustache</code> tag. Error thrown when unescaped templating syntax found in an attribute in another tag.</td>
+  </tr>
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#attribute-contains-template-partial">TEMPLATE_PARTIAL_IN_ATTR_VALUE</a></td>
```
Comment (Gregable) [inline extensions/amp-mustache/amp-mustache.md]: A useful reference link might be: https://github.com/janl/mustache.js/#partials

File: extensions/amp-mustache/amp-mustache.md [hunk 9]
```diff
@@ -16,7 +16,50 @@ limitations under the License.
...
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#tag-required-by-another-tag-is-missing">TAG_REQUIRED_BY_MISSING</a></td>
+    <td>Error thrown when required <code>amp-mustache</code> extension <code>.js</code> script tag is missing or incorrect.</td>
+  </tr>
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#attribute-contains-template-syntax">TEMPLATE_IN_ATTR_NAME</a></td>
+    <td>Mustache templates must be within the <code>amp-mustache</code> tag. Error thrown when templating syntax found in an attribute in another tag.</td>
+  </tr>
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#attribute-contains-unescaped-template-syntax">UNESCAPED_TEMPLATE_IN_ATTR_VALUE</a></td>
```
Comment (Gregable) [inline extensions/amp-mustache/amp-mustache.md]: A useful reference link for "unescaped templating syntax" might be https://github.com/janl/mustache.js/#variables

File: extensions/amp-install-serviceworker/amp-install-serviceworker.md [hunk 10]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+    <td>None</td>
+  </tr>
+</table>
+
+The following lists validation errors specific to the `amp-install-serviceworker` tag
+(see also `amp-install-serviceworker` in the [AMP validator specification](https://github.com/ampproject/amphtml/blob/master/validator/validator.protoascii):
+
+<table>
+  <tr>
+    <th width="40%"><strong>Validation Error</strong></th>
```
Comment (Gregable) [inline extensions/amp-install-serviceworker/amp-install-serviceworker.md]: Your docs made me notice that validation should require `src` and a NODISPLAY `layout`. I'm making that change now. You may want to document those here too.

**Comment by Gregable** [comment]
Looks good, this must have been a lot of work! The only other high level comment is that we've changed the rendering of the errors to no longer display the error code, but rather a human-readable string. Probably not in this pull request, but we may want to revisit the format of these possible validation error tables down the line.

**Comment by cramforce** [comment]
Thanks @Meggin! This no longer matches the actual output, right? Should this maybe go directory into the `protoascii` file and we'd just generated the `.md` from there?

File: extensions/amp-audio/amp-audio.md [hunk 11]
```diff
@@ -16,13 +16,60 @@ limitations under the License.
 
 ### <a name="amp-audio"></a> `amp-audio`
 
-A replacement for the HTML5 `audio` tag. Like all embedded external resources in a AMP file, the audio is lazily loaded only when the `amp-audio` element is in or near the viewport.
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>A replacement for the HTML5 `audio` tag. The `amp-audio` component is only to be used for direct HTML5 audio file embeds.</td>
```
Comment (rudygalfi) [inline extensions/amp-audio/amp-audio.md]: Here, audio and amp-audio just appear with ` around them. Did you mean to use <code>?

**Comment by Meggin** [comment]
We should absolutely automate this stuff.  And I believe that work is
happening. The hope is this will hold us over till that can happen.

Also there's a lot to learn from the exercise of trying to do this that
could hopefully feed into the automation.

So yeah. How can I help with automation?

File: extensions/amp-brightcove/amp-brightcove.md [hunk 12]
```diff
@@ -16,9 +16,61 @@ limitations under the License.
 
 ### <a name="amp-brightcove"></a> `amp-brightcove`
 
-An `amp-brightcove` component displays a Brightcove [Video Cloud](https://www.brightcove.com/en/online-video-platform) or [Perform](https://www.brightcove.com/en/perform) player.
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>An `amp-brightcove` component displays a Brightcove [Video Cloud](https://www.brightcove.com/en/online-video-platform) or [Perform](https://www.brightcove.com/en/perform) player.</td>
```
Comment (rudygalfi) [inline extensions/amp-brightcove/amp-brightcove.md]: \<code\>amp-brightcove\</code\>?

File: extensions/amp-facebook/amp-facebook.md [hunk 13]
```diff
@@ -16,9 +16,44 @@ limitations under the License.
...
 
-**Status: Component landed, but does not validate.**
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>Displays a Facebook Post or Video. </td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Availability</strong></td>
+    <td>Stable, but doesn't validate.</td>
```
Comment (rudygalfi) [inline extensions/amp-facebook/amp-facebook.md]: This line will need to be updated per @Gregable's comment below.

File: extensions/amp-install-serviceworker/amp-install-serviceworker.md [hunk 14]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
 
 ### <a name="amp-install-serviceworker"></a> `amp-install-serviceworker`
 
-The `amp-install-serviceworker` component allows installing a [ServiceWorker](http://www.html5rocks.com/en/tutorials/service-worker/introduction/) for the current page.
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>Installs a [ServiceWorker](http://www.html5rocks.com/en/tutorials/service-worker/introduction/) for the current page.</td>
```
Comment (rudygalfi) [inline extensions/amp-install-serviceworker/amp-install-serviceworker.md]: The markdown syntax appears directly here when within HTML code.

File: extensions/amp-install-serviceworker/amp-install-serviceworker.md [hunk 15]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+  </tr>
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#tag-required-by-another-tag-is-missing">TAG_REQUIRED_BY_MISSING</a></td>
+    <td>Error thrown when required <code>amp-install-serviceworker</code> extension <code>.js</code> script tag is missing or incorrect.</td>
+  </tr>
+  <tr>
+</table>
 
-The idea here is that this ServiceWorker runs whenever the AMP file is served from the origin where you publish the AMP file. The ServiceWorker will not be loaded when the document is loaded from an AMP cache.
+### Behavior
```
Comment (rudygalfi) [inline extensions/amp-install-serviceworker/amp-install-serviceworker.md]: Should be 4th level, "####"

File: extensions/amp-install-serviceworker/amp-install-serviceworker.md [hunk 16]
```diff
@@ -33,10 +69,6 @@ See [this article](https://medium.com/@cramforce/amps-and-websites-in-the-age-of
 
 ```
 
-### Behavior
```
Comment (rudygalfi) [inline extensions/amp-install-serviceworker/amp-install-serviceworker.md]: Should be 4th level, "####"

File: extensions/amp-install-serviceworker/amp-install-serviceworker.md [hunk 17]
```diff
@@ -33,10 +69,6 @@ See [this article](https://medium.com/@cramforce/amps-and-websites-in-the-age-of
...
-
-Registers the ServiceWorker given by the `src` attribute. If the current origin is different from the origin of the ServiceWorker, this custom element does nothing (Emits warning in development mode).
-
 ### Attributes
```
Comment (rudygalfi) [inline extensions/amp-install-serviceworker/amp-install-serviceworker.md]: Should be 4th level, "####"

File: extensions/amp-install-serviceworker/amp-install-serviceworker.md [hunk 18]
```diff
@@ -33,10 +69,6 @@ See [this article](https://medium.com/@cramforce/amps-and-websites-in-the-age-of
...
 
 #### `src`
```
Comment (rudygalfi) [inline extensions/amp-install-serviceworker/amp-install-serviceworker.md]: Should be 5th level, "#####" (etc. if there are more)

File: extensions/amp-list/amp-list.md [hunk 19]
```diff
@@ -19,6 +19,54 @@ limitations under the License.
 The `amp-list` fetches the dynamic content from a CORS JSON endpoint and renders it
 using a supplied template.
 
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>Displays an iframe.</td>
```
Comment (rudygalfi) [inline extensions/amp-list/amp-list.md]: Update to reflect amp-list behavior.

File: extensions/amp-list/amp-list.md [hunk 20]
```diff
@@ -19,6 +19,54 @@ limitations under the License.
...
+  </tr>
+  <tr>
+    <td width="40%"><strong>Availability</strong></td>
+    <td>Stable</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Required Script</strong></td>
+    <td><code>&lt;script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js">&lt;/script></code></td>
```
Comment (rudygalfi) [inline extensions/amp-list/amp-list.md]: should say amp-list

File: extensions/amp-user-notification/amp-user-notification.md [hunk 21]
```diff
@@ -16,14 +16,50 @@ limitations under the License.
...
+    <td width="40%"><strong>Description</strong></td>
+    <td>Displays a dismissable notification to the user. </td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Availability</strong></td>
+    <td>Stable</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Required Script</strong></td>
+    <td><code>&lt;script async custom-element="amp-font" src="https://cdn.ampproject.org/v0/amp-font-0.1.js">&lt;/script></code></td>
```
Comment (rudygalfi) [inline extensions/amp-user-notification/amp-user-notification.md]: should say amp-user-notification

File: extensions/amp-analytics/amp-analytics.md [hunk 22]
```diff
@@ -16,7 +16,43 @@ limitations under the License.
 
 # <a name="amp-analytics"></a>`amp-analytics`
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: Make level 3: "###"

File: extensions/amp-analytics/amp-analytics.md [hunk 23]
```diff
@@ -16,7 +16,43 @@ limitations under the License.
...
+    <th width="40%"><strong>Validation Error</strong></th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td width="40%"><a href="/docs/reference/validation_errors.html#tag-required-by-another-tag-is-missing">TAG_REQUIRED_BY_MISSING</a></td>
+    <td>Error thrown when required <code>amp-analytics</code> extension <code>.js</code> script tag is missing or incorrect.</td>
+  </tr>
+</table>
 
 ## <a name="behavior"></a>Behavior
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: Make level 4: "####". Restructure the rest accordingly.

**Comment by rudygalfi** [comment]
Spotted a few errors and left comments. Overall LGTM once those are addressed.

**Comment by rudygalfi** [comment]
Any updates on revisions to this?

**Comment by rudygalfi** [comment]
LGTM on the latest revisions

**Comment by Meggin** [comment]
PTAL. Hoping this can be merged.

**Comment by rudygalfi** [comment]
Unfortunately out of sync with base now, but agree this can be merged.

cc @cramforce

**Comment by Meggin** [comment]
Closing this one, as I created a clean PR that's up-to-date with master here: https://github.com/ampproject/amphtml/pull/2048

Which Greg just kicked off merge.

**Pull request closed without merging**